packages/blocks/src/store/reducer.js

```diff
@@ -43,6 +43,47 @@ export function blockTypes( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Reducer managing the block style variations.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function blockStyles( state = {}, action ) {
+	switch ( action.type ) {
+		case 'ADD_BLOCK_TYPES':
```

Comment:
I didn't explicitly because the registration of styles can happen even before the block exists or after, that's one of the main ideas of this PR. You can register your styles and your block whenever you want, the styles will always be there.

For example, a plugin author could unregister a block type, register it back slightly modified and the styles will still be there.

Reply:
Interesting way of thinking. The issue I see here is that you could add styles with block X and then remove it but its styles will be still there.

Reply:
Is this really an issue? I mean aside some state being there and not really useful.

Reply:
It would be if you reuse the same block name to register your own version of the block, it would have also styles from the old block. Not a big deal in my opinion, but it raises a flag if you would scale the pattern for other parts of the block settings. I'm thinking how to turn `block.registerBlockType` hook into a self-dispatching action which does the same what you did here :)